crates/eth-rpc/src/servers/alchemy_rpc.rs

```diff
@@ -21,8 +29,65 @@ impl<P: Provider + Send + Sync> AlchemyRpc<P> {
 
 #[async_trait]
 impl<P: Provider + Send + Sync + 'static> AlchemyApiServer for AlchemyRpc<P> {
+    async fn token_allowance(
+        &self,
+        contract_address: Address,
+        account_address: Address,
+        spender_address: Address,
+    ) -> Result<TokenAllowance> {
+        let token_allowance =
+            self.kakarot_client.token_allowance(contract_address, account_address, spender_address).await?;
+        Ok(token_allowance)
+    }
+
     async fn token_balances(&self, address: Address, contract_addresses: Vec<Address>) -> Result<TokenBalances> {
         let token_balances = self.kakarot_client.token_balances(address, contract_addresses).await?;
         Ok(token_balances)
     }
+
+    async fn token_metadata(&self, contract_address: Address) -> Result<TokenMetadata> {
+        let selectors = vec!["decimals()", "name()", "symbol()"];
```

Comment:
should use kakarot_client here right?

Reply:
yes call should be forwarded to kakarot_client